### PR TITLE
Feature - Change Remaining References to Lead School

### PR DESCRIPTION
**Pull request by adamjcolvin**
### Context
[Trello](https://trello.com/c/z4eDMVkv)

We need to rename every reference to `lead_school` and its variations to `lead_partner`. This will be a big change, so let's use this branch as a feature branch until we're ready. 

### Changes proposed in this pull request
Rename all `lead_school` references to `lead_partner`, fixing any broken tests and removing code where it makes sense. 

### Important business

- [ ] Does this PR introduce any PII fields that need to be overwritten or deleted in db/scripts/sanitise.sql?
- [ ] Does this PR change the database schema? If so, have you updated the config/analytics.yml file and considered whether you need to send 'import_entity' events?
- [ ] Do we need to send any updates to DQT as part of the work in this PR?
- [ ] Does this PR need an ADR?

NB: Please notify the #twd_data_insights team and ask for a review if new fields are being added to analytics.yml

### CODE REVIEW

File: app/lib/funding/parsers/lead_partner_trainee_summaries.rb [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 module Funding
   module Parsers
-    class LeadSchoolTraineeSummaries < Base
+    class LeadPartnerTraineeSummaries < Base
       class << self
         def id_column
           "Lead school URN"
```
Comment (kennyevil) [inline app/lib/funding/parsers/lead_partner_trainee_summaries.rb:8]: Do we need to change this column name?

File: app/services/funding/lead_partner_payment_schedules_importer.rb [hunk 2]
```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 module Funding
-  class LeadSchoolPaymentSchedulesImporter < PayablePaymentSchedulesImporter
+  class LeadPartnerPaymentSchedulesImporter < PayablePaymentSchedulesImporter
     def payable(id)
       School.find_by(urn: id)
```
Comment (kennyevil) [inline app/services/funding/lead_partner_payment_schedules_importer.rb:6]: Should this be changed to `LeadPartner.find_by(urn: id)`?

**Review by kennyevil — COMMENTED** [review]
Looks good. Couple of questions, but they might be addressed in the subsequent PR.

**Comment by adamjcolvin** [comment]
Thanks @kennyevil, you're right on both counts, but they'll be addressed in later PRs. I'm happy to keep this open, so we can see all the changes together before merging to main. Or maybe it's best to merge now, what do you think?

**Review by kennyevil — APPROVED** [review]

**Comment by kennyevil** [comment]
@adamjcolvin Let's merge it now, you've got the second one open so it can be sorted out there.

**Pull request merged**